test/src/code_field/code_controller_indent_selection_test.dart

```diff
@@ -0,0 +1,477 @@
+
+    test('description', () {
+      final examples = [
+        _Example(
+          name: 'Indentation through a folded block affects and unfolds it',
+          initialFullText: '''
+class MyClass {
+  void getSmth(){
+
+  }
+
+  void setSmth(){
+
+  }
+}
+''',
+          initialVisibleText: '''
+class MyClass {
+  void getSmth(){
+
+  void setSmth(){
+
+  }
+}
+''',
+          expectedFullText: '''
+${indent}class MyClass {
+$indent  void getSmth(){
+
+$indent  }
+
+$indent  void setSmth(){
+
+  }
+}
+''',
+          expectedVisibleText: '''
+${indent}class MyClass {
+$indent  void getSmth(){
+
+$indent  }
+
+$indent  void setSmth(){
+
+  }
+}
+''',
+          foldableBlockIndex: 1,
+          initialSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 44,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 0 + indentLength,
+            extentOffset: 44 + indentLength * 4 + 5, // 5 -> hidden
+          ),
+        ),
+        _Example(
+          name: 'Indentation before folded blocks does not affect neither open them',
+          initialFullText: '''
+class MyClass {
+  void getSmth(){
+
+  }
+
+  void setSmth(){
+
+  }
+}
+''',
+          initialVisibleText: '''
+class MyClass {
+  void getSmth(){
+
+  }
+
+  void setSmth(){
+}
+''',
+          expectedFullText: '''
+${indent}class MyClass {
+$indent  void getSmth(){
+
+$indent  }
+
+  void setSmth(){
+
+  }
+}
+''',
+          expectedVisibleText: '''
+${indent}class MyClass {
+$indent  void getSmth(){
+
+$indent  }
+
+  void setSmth(){
+}
+''',
+          foldableBlockIndex: 2,
+          initialSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 38,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 0 + indentLength,
+            extentOffset: 38 + indentLength * 3,
+          ),
+        ),
+      ];
+
+      for (final example in examples) {
+        controller.text = example.initialFullText;
+        controller.foldAt(controller.code.foldableBlocks[example.foldableBlockIndex!].firstLine);
+        controller.selection = example.initialSelection;
+
+        expect(
+          controller.value.text,
+          example.initialVisibleText,
+          reason: 'assertion of an arranged data',
+        );
+
+        controller.indentSelection();
+
+        expect(
+          controller.value.text,
+          example.expectedVisibleText,
+          reason: example.name,
+        );
+        expect(
+          controller.code.text,
+          example.expectedFullText,
+          reason: example.name,
+        );
+        expect(
+          controller.value.selection,
+          example.expectedSelection,
+          reason: example.name,
+        );
+      }
+    });
+  });
+}
+
+class _Example {
+  final String name;
+  final String initialFullText;
+  final String? initialVisibleText;
+  final String expectedFullText;
+  final String expectedVisibleText;
```

Comment:
You may just test the full text and that the folded block indexes are the same before and after the test.